src/core_plugins/elasticsearch/lib/kibana_index_mappings.js

```diff
@@ -1,16 +1,36 @@
-export const mappings = {
-  config: {
-    properties: {
-      buildNum: {
-        type: 'keyword'
-      }
-    }
-  },
-  server: {
-    properties: {
-      uuid: {
-        type: 'keyword'
+import _ from 'lodash';
+
+class KibanaMappings {
+  constructor(plugin) {
+    this._plugin = plugin;
+    this._defaultMappings = {
+      '_default_': {
+        'dynamic': 'strict'
+      },
+      config: {
+        dynamic: true,
```

Comment:
I was totally blanking on the fact that the advanced settings get stored in this doc type. This makes sense now.